Implement RefArch Access / Set up AWS Auth (#798)
gruntwork-io/docs: _docs-sources/refarch/access/how-to-auth-vpn/index.md

# VPN Authentcation 

Most of the AWS resources that comprise the Reference Architecture run in private subnets, which means they do not have a public IP address, and cannot be reached directly from the public Internet. This reduces the "surface area" that attackers can reach. Of course, you still need access into the VPCs, so we exposed a single entrypoint into the network: an [OpenVPN server](https://openvpn.net/).

> You use an OpenVPN client to connect to the OpenVPN server, which gets you "in" to the network, allowing you to securely connect to private resources in the account.

Reviewer: ```suggestion
You use an OpenVPN client to connect to the OpenVPN server. Once connected to the VPN, you can securely connect to private resources in the account.
```

> Here are the steps you'll need to take:

Reviewer: Can we re-write this list as a sentence/paragraph, since the sections speak for themselves and directly below?

Reviewer 2: Resolved in 62fe8df61a4e5f904e27d312227fa5acb35af668

1. [Install an OpenVPN client](#install-an-openvpn-client)
1. [Join the OpenVPN IAM Group](#join-the-openvpn-iam-group)
1. [Use openvpn-admin to generate a configuration file](#use-openvpn-admin-to-generate-a-configuration-file)
1. [Connect to the OpenVPN server](#connect-to-the-openvpn-server)

## Install an OpenVPN client

> There are free and paid OpenVPN clients available for most major operating systems:

Reviewer 2: Agreed, addressed in f2e7940

1. OS X: [Viscosity](https://www.sparklabs.com/viscosity/) or [Tunnelblick](https://tunnelblick.net/).
1. Windows: [official client](https://openvpn.net/index.php/open-source/downloads.html).
1. Linux: `apt-get install openvpn` or `yum install openvpn`.

## Join the OpenVPN IAM Group

> Your IAM User needs access to SQS queues used by the OpenVPN server. Since IAM users are defined only in the security account, and the OpenVPN servers are defined in separate AWS accounts (stage, prod, etc), that means you need to "switch" to the accounts with the OpenVPN servers by assuming an IAM Role that has access to the SQS queues in those accounts.

Reviewer: ```suggestion
Your IAM User needs access to SQS queues used by the OpenVPN server. Since IAM users are defined only in the security account, and the OpenVPN servers are defined in separate AWS accounts (stage, prod, etc), that means you need to authenticate to the accounts with the OpenVPN servers by assuming an IAM Role that has access to the SQS queues in those accounts.
```

> To be able to assume an IAM Role, your IAM user needs to be part of an IAM Group with the proper permissions, such as `_account.xxx-full-access` or `_account.xxx-openvpn-users`, where `xxx` is the name of the account you want to access (stage, prod, etc). See Configure other IAM users for instructions on adding users to IAM Groups.

Reviewer: Should we cross link to `Configure other IAM users`?

Reviewer 2: Resolved in 7e60b8e158e71f8ed9a0aa51672ef45af3f1354f
